opm/simulators/flow/BlackoilModelEbosNldd.hpp

```diff
@@ -616,30 +647,25 @@ class BlackoilModelEbosNldd {
             for (int ii : {0, 1}) {
                 if (std::isnan(res[ii])) {
                     report.setReservoirFailed({types[ii], CR::Severity::NotANumber, compIdx});
-                    if (model_.terminalOutputEnabled()) {
-                        OpmLog::debug("NaN residual for " + model_.compNames().name(compIdx) + " equation.");
-                    }
+                    logger.debug("NaN residual for " + model_.compNames().name(compIdx) + " equation.");
                 } else if (res[ii] > model_.param().max_residual_allowed_) {
                     report.setReservoirFailed({types[ii], CR::Severity::TooLarge, compIdx});
-                    if (model_.terminalOutputEnabled()) {
-                        OpmLog::debug("Too large residual for " + model_.compNames().name(compIdx) + " equation.");
-                    }
+                    logger.debug("Too large residual for " + model_.compNames().name(compIdx) + " equation.");
                 } else if (res[ii] < 0.0) {
                     report.setReservoirFailed({types[ii], CR::Severity::Normal, compIdx});
-                    if (model_.terminalOutputEnabled()) {
-                        OpmLog::debug("Negative residual for " + model_.compNames().name(compIdx) + " equation.");
-                    }
+                    logger.debug("Negative residual for " + model_.compNames().name(compIdx) + " equation.");
                 } else if (res[ii] > tol[ii]) {
                     report.setReservoirFailed({types[ii], CR::Severity::Normal, compIdx});
                 }
             }
         }
 
         // Output of residuals.
-        if (model_.terminalOutputEnabled())
-        {
-            // Only rank 0 does print to std::cout
+        if (report.converged() && iteration == 0) {
+            // Log nothing.
+        } else {
```

Comment:
It's probably just me, but I typically prefer to write
```C++
if (some_condition) {
    // Empty block
} else {
    // Some logic
}
```
as
```C++
if (! some_condition) {
    // Some logic
}
```
instead.  In this case it'd be
```C++
if (! report.converged() || (iteration != 0)) {
    // ...
}
```